shm/include/ship.hpp

```diff
@@ -8,9 +8,14 @@
 
 class Ship {
 public:
+    class Delegate {
+        public:
+        virtual void payCrew(size_t crew_count) = 0;
+        virtual ~Delegate(){};
+    };
```

Comment:
Why is this class in Ship class?

Reply:
I think that this is how the delegate pattern works, according to what Mateusz has recently shown us on the lecture Obiektowosc#3. If I remember correctly in his class Ship he implemented nested class Delegate (after which Player inherits) with only one pure virtual func that Ship class needs to perform nextDay() func. But maybe I got something wrong, I am not certain.